-pictureInPicturePlayerLayerView is unavailable on watchOS

jernoble opened the pull request:
#### 139d1d031bd3f24cbf275dd3a33d5722ddbd3cdf
<pre>
-pictureInPicturePlayerLayerView is unavailable on watchOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=250910">https://bugs.webkit.org/show_bug.cgi?id=250910</a>
rdar://104475956

Reviewed by Chris Dumez and Eric Carlson.

Add a HAVE_PICTUREINPICTUREPLAYERLAYERVIEW guard and use it liberally in VideoFullscreenInterfaceAVKit.mm.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(WebAVPlayerLayerView_dealloc):
(allocWebAVPlayerLayerViewInstance):
(VideoFullscreenInterfaceAVKit::videoDimensionsChanged):
(VideoFullscreenInterfaceAVKit::doEnterFullscreen):

Canonical link: <a href="https://commits.webkit.org/259189@main">https://commits.webkit.org/259189@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/06183a2456d837769e25b8172b5d906dc827ab90

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104099 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13190 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37022 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113308 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173612 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108041 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14256 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4106 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96329 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112381 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109869 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10986 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94048 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38655 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92837 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25660 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80311 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/94165 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6558 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27036 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/90684 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/4341 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6700 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3574 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29910 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12700 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46572 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/99295 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6332 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8478 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/24978 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson reviewed (approved):
r=me

cdumez commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1400,3 +1400,7 @@
 #endif
+
+#if PLATFORM(IOS)
+#define HAVE_PICTUREINPICTUREPLAYERLAYERVIEW
```
`#define HAVE_PICTUREINPICTUREPLAYERLAYERVIEW 1` ?

cdumez reviewed (approved):
LGTM too, although I would have used more underscores in `HAVE_PICTUREINPICTUREPLAYERLAYERVIEW` myself to help with readability a little :)

jernoble committed:
-pictureInPicturePlayerLayerView is unavailable on watchOS
https://bugs.webkit.org/show_bug.cgi?id=250910
rdar://104475956

Reviewed by Chris Dumez and Eric Carlson.

Add a HAVE_PICTUREINPICTUREPLAYERLAYERVIEW guard and use it liberally in VideoFullscreenInterfaceAVKit.mm.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(WebAVPlayerLayerView_dealloc):
(allocWebAVPlayerLayerViewInstance):
(VideoFullscreenInterfaceAVKit::videoDimensionsChanged):
(VideoFullscreenInterfaceAVKit::doEnterFullscreen):

Canonical link: https://commits.webkit.org/259189@main
